### PR TITLE
Provide human-friendly error if invalid class is specified for a command

### PR DESCRIPTION
**Pull request by szepeviktor**
When non-existent class given to `add_command()`.

### CODE REVIEW

**Comment by szepeviktor** [comment]
Learned `git rebase -i HEAD~3`

**Comment by danielbachhuber** [comment]
@szepeviktor looks like you need to update your use of `class_exists()`

**Comment by szepeviktor** [comment]
`WP_CLI::add_command( 'import', new Import_Command );` in `import.php` is out of *my* understanding. PHPDoc says `@param string $class The command implementation`.

**Comment by szepeviktor** [comment]
The `class Import_Command` has no constructor. What does `new` do?

**Comment by szepeviktor** [comment]
1. Should I change `import.php' and check `$class` using `is_string()`
2. Or something else I do not see yet?

File: php/class-wp-cli.php [hunk 1]
```diff
@@ -166,6 +166,10 @@ public static function do_hook( $when ) {
 	 *   'before_invoke' => callback to execute before invoking the command
 	 */
 	public static function add_command( $name, $class, $args = array() ) {
+		if ( ! class_exists( $class ) ) {
```
Comment (danielbachhuber) [inline php/class-wp-cli.php]: You can skip this check if the class is already instantiated so:

```
if ( is_string( $class ) && ! class_exists( $class ) ) {
```

**Comment by szepeviktor** [comment]
All green.

**Comment by danielbachhuber** [comment]
@szepeviktor can we have a test too?

**Comment by szepeviktor** [comment]
Special case!
Is it OK?

```feature
Feature: WP-CLI Commands

  Scenario: Invalid class is specified for a command
    Given an empty directory
    And a custom-cmd.php file:
      """
      <?php

      WP_CLI::add_command( 'command example', 'Non_Existent_Class' );
      """
    And I run `wp --require=custom-cmd.php help`

    When I run `wp help`
    Then STDOUT should contain:
      """
      Error: Class 'Non_Existent_Class' does not exist.
      """
```

**Comment by danielbachhuber** [comment]
Yep, looks good

**Comment by szepeviktor** [comment]
Could you finish it? May behat test failed here

```behat
When I run `wp --require=custom-cmd.php help`    # features/steps/when.php:18
  $ wp --require=custom-cmd.php help

  Error: Class 'Non_Existent_Class' does not exist.
  cwd: /tmp/wp-cli-test-run-54de89a8752648.04528450/
  exit status: 1
```

File: features/command.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  Scenario: Invalid class is specified for a command
+    Given an empty directory
+    And a custom-cmd.php file:
+      """
+      <?php
+
+      WP_CLI::add_command( 'command example', 'Non_Existent_Class' );
+      """
+
+    When I run `wp --require=custom-cmd.php help`
```
Comment (danielbachhuber) [inline features/command.feature]:
```
When I try ...
Then STDERR should contain
```

**Comment by szepeviktor** [comment]
I am trying with

```
When I run `wp --require=custom-cmd.php help`
Then STDERR should contain:
  """
  Class 'Non_Existent_Class' does not exist.
  """
```

**Comment by szepeviktor** [comment]
Travis: Just the *usual*. Consider it to be all green.

**Pull request merged**